app/models/cafe_work.rb

```diff
@@ -0,0 +1,248 @@
+# encoding: UTF-8
+class CafeWork < ActiveRecord::Base
+  # Associations
+  belongs_to :profile
+  has_and_belongs_to_many :councils
+
+  # Validations
+  validates :work_day, :pass, :lp, :lv, presence: true
+  validates :pass, :lp, inclusion: { in: 1..4 }
+  validates :lv, inclusion: { in: 1..7 }
+  validates :name, :lastname, :phone, :email, presence: true, if: :has_worker?
+  validates_uniqueness_of :pass, scope: [:work_day, :lv, :lp, :d_year]
```

Comment:
Prefer the new style validations `validates :column, uniqueness: value` over `validates_uniqueness_of`.